ui/client/components/graph/node-modal/editors/expression/EditableExpression.js

```diff
@@ -0,0 +1,71 @@
+import RawEditor from "./RawEditor"
+import BoolEditor from "./BoolEditor"
+import React from "react"
+import ExpressionWithFixedValues from "./ExpressionWithFixedValues"
+import {parseableBoolean} from "./ExpressionParser"
+
+export default class EditableExpression extends React.Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      displayRawEditor: true
+    }
+  }
+
+  render() {
+    const {fieldType, expressionObj, rowClassName, valueClassName, showSwitch} = this.props
+    const editorName = resolveEditorName(fieldType, expressionObj, this.state.displayRawEditor)
+    const Editor = editorTypes[editorName]
+    return <Editor toggleEditor={this.toggleEditor}
+                   switchable={this.switchable(editorName, expressionObj)}
+                   shouldShowSwitch={this.showSwitch(fieldType, showSwitch)}
+                   rowClassName={rowClassName ? rowClassName : "node-row"}
+                   valueClassName={valueClassName ? valueClassName : "node-value"}
+                   {...this.props}/>
+  }
+
+  toggleEditor = (_) => this.setState({
+    displayRawEditor: !this.state.displayRawEditor
+  })
+
+  switchable = (editorName, expressionObj) => {
+    switch (editorName) {
+      case boolEditor:
+        return true
+      case rawEditor:
+        return parseableBoolean(expressionObj)
+      default:
+        return false
+    }
+  }
+
+  showSwitch = (fieldType, showSwitch) => showSwitch && supportedFieldTypes.includes(fieldType)
+}
+
+const resolveEditorName = (fieldType, expressionObj, displayRawEditor) => {
```

Comment:
Why you use once time stiing, other time defined value? There should be something like:

```
const EditorType = {
  BOOL_EDITOR: "boolEditor",
  RAW_EDITOR: "rawEditor",
 EXPRESSION: "expression",
 BOOLEAN: "Boolean",
 EXPRESSION_WITH_FIXED_VALUE: "expressionWithFixedValues",

supportedFieldTypes: ["expression", "Boolean"],

isSupported = (filedType) => this.supportedFieldTypes.includes(fieldType),

else methods correlated with fieldTypes.. 
}
```

Reply:
Done